src/com/sap/piper/DefaultValueCache.groovy

```diff
@@ -2,6 +2,19 @@ package com.sap.piper
 
 import com.cloudbees.groovy.cps.NonCPS
 
+/*
+ * Provides the default values in an immutable form. Maps, Sets, Lists
+ * contained nested in the outermost Map are also immutable.
+ *
+ * java.util.Date instances are refused since they are not immutable.
+ *
+ * All other Objects we expect here (... basically from parsing json) are expected to be
+ * immutable:
+ *   - java.lang.String (in the unlikely case we see other CharSequences we convert to String)
```

Comment:
What about GStringImpl? Do we need that also?

Reply:
We do not get `GString`s when parsing json files. See e.g. [here](http://docs.groovy-lang.org/docs/next/html/documentation/json-userguide.html).